pkg/pulumiyaml/run.go

```diff
@@ -855,8 +885,27 @@ func (ctx *evalContext) registerResource(kvp resourceNode) (lateboundResource, b
 		} else {
 			overallOk = false
 		}
+	} else if defaultPkgInfo, ok := ctx.defaultPackageInfo[pkgName]; ok && !strings.HasPrefix(v.Type.Value, "pulumi:providers:") && false {
+		// if no provider set, copy default provider to the resource
+		expr, _ := ast.Interpolate("${" + defaultPkgInfo.providerResource.Value + "}")
+		providerOpt, ok := ctx.evaluateResourceValuedOption(expr, "provider")
```

Comment:
This isn't quite working, as `providerOpt` resolves to a string 
    error: expected resource, got string
    error: error reading from server: EOF
 
```
    panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x104a87b00]
    goroutine 83 [running]:
    github.com/pulumi/pulumi-yaml/pkg/pulumiyaml/ast.(*StringExpr).Syntax(0x1400091c1a0?)
        <autogenerated>:1
    github.com/pulumi/pulumi-yaml/pkg/pulumiyaml/ast.ExprError({0x104eb62d8?, 0x0?}, {0x14000462030, 0x30}, {0x0, 0x0})
        /Users/alexqiu/Desktop/pulumi-yaml/pkg/pulumiyaml/ast/expr.go:44 +0x4c
    github.com/pulumi/pulumi-yaml/pkg/pulumiyaml.(*evalContext).error(0x104ae0c83?, {0x104eb62d8?, 0x0?}, {0x14000462030?, 0x1?}) 
```